src/examples/crypto/rsa/rsa.ts

```diff
@@ -160,41 +160,7 @@ function rangeCheck116(x: Field) {
  */
 function rangeCheck128Signed(xSigned: Field) {
   let x = xSigned.add(1n << 127n);
-
-  let [x0, x1] = Provable.witnessFields(2, () => [
-    x.toBigInt() & ((1n << 64n) - 1n),
-    x.toBigInt() >> 64n,
-  ]);
-
-  Gadgets.rangeCheck64(x0);
-  Gadgets.rangeCheck64(x1);
-
-  x0.add(x1.mul(1n << 64n)).assertEquals(x);
-}
-
-let rsa = ZkProgram({
-  name: 'rsa-verify',
-
-  methods: {
-    verify: {
-      privateInputs: [Bigint2048, Bigint2048, Bigint2048],
-
-      async method(
-        message: Bigint2048,
-        signature: Bigint2048,
-        modulus: Bigint2048
-      ) {
-        rsaVerify65537(message, signature, modulus);
-      },
-    },
-  },
-});
-
-let { verify } = await rsa.analyzeMethods();
-
-console.log(verify.summary());
-console.log('rows', verify.rows);
-
-tic('compile');
-await rsa.compile();
-toc();
+  Gadgets
+    .isDefinitelyInRangeN(128, x)
+    .assertTrue("BigInt carry should not exceed 128 bits!");
```

Comment:
I think this should be changed back, `isDefinitelyInRangeN()` uses a different gate which uses more constraints than `rangeCheck64()`, and the extra creation of a `Bool` instead of asserting directly also wastes constraints

Reply:
I am happy to learn about gates intricacies.
True - this change alone added 4000 rows :open_mouth: 
- Reverting the changes causes a weird `Snarky.run` error :thinking: 
- I did a workaround by using `Provable.witness(Provable.Array(Field, 2), () => {})` instead of `witnessFields(2, callback)`
- Now everything pass and  rsa has the following summary:
   -  Total rows: 12401 
   - RangeCheck0: 2488
   -  Generic: 9913